### PR TITLE
[libclc] Add new function implementations in math.

### PR DESCRIPTION
**Pull request by GYDmedwin**
Add some new function implementations in the math package, so that the operator can run normally

### CODE REVIEW

File: libclc/amdgcn-amdhsa/libspirv/math/cos.cl [hunk 1]
```diff
@@ -6,9 +6,19 @@
...
+//#include <spirv/spirv.h>
 
-_CLC_OVERLOAD _CLC_DECL _CLC_CONSTFN __clc_fp32_t
-__spirv_ocl_cos(__clc_fp32_t In) {
-  return __builtin_amdgcn_cosf(In);
-}
+//_CLC_OVERLOAD _CLC_DECL _CLC_CONSTFN __clc_fp32_t
+//__spirv_ocl_cos(__clc_fp32_t In) {
+//  return __builtin_amdgcn_cosf(In);
+//}
```
[the remark this answers is not in the document]
Comment (zjin-lcf) [reply libclc/amdgcn-amdhsa/libspirv/math/cos.cl:14]: @GYDmedwin 
Could the author explain that the builtin cosine function is replaced with another cosine function ?  Thanks.

[the remark this answers is not in the document]
Comment (GYDmedwin) [reply libclc/amdgcn-amdhsa/libspirv/math/cos.cl:14]: We conducted a benchmark test. The original plan did not compile and passed, and the new modification can make the benchmark compile and run normally.

File: libclc/amdgcn-amdhsa/libspirv/SOURCES [hunk 2]
```diff
@@ -1,17 +1,57 @@
...
+math/logb.cl
+math/log2.cl
+math/log1p.cl
+math/rsqrt.cl
 workitem/get_global_size.cl
 workitem/get_local_size.cl
 workitem/get_num_groups.cl
 workitem/get_max_sub_group_size.cl
 workitem/get_num_sub_groups.cl
 workitem/get_sub_group_id.cl
```
[the remark this answers is not in the document]
Comment (GYDmedwin) [reply libclc/amdgcn-amdhsa/libspirv/SOURCES:56]: Sorry, the wrong file was selected when submitting, new submission has modified this.

File: libclc/amdgcn-amdhsa/libspirv/math/frexp.inc [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  return res;
+}
+
+#define __CLC_VECTORIZE_FREXP_IMPL(GENTYPE, VEC_LEN, HALF_VEC_LEN)             \
+  _CLC_OVERLOAD _CLC_DEF GENTYPE __spirv_ocl_frexp(                            \
+      GENTYPE x, __CLC_ADDRESS_SPACE int##VEC_LEN *ep) {                       \
+    int##HALF_VEC_LEN ep_lo;                                                   \
+    int##HALF_VEC_LEN ep_hi;                                                   \
+    GENTYPE res = (GENTYPE)(__spirv_ocl_frexp(x.lo, &ep_lo),                   \
+                            __spirv_ocl_frexp(x.hi, &ep_lo));                  \
```
[the remark this answers is not in the document]
Comment (GYDmedwin) [reply libclc/amdgcn-amdhsa/libspirv/math/frexp.inc:50]: Thank you!

File: libclc/amdgcn-amdhsa/libspirv/math/log2.cl [hunk 4]
```diff
@@ -0,0 +1,43 @@
+ #include <spirv/spirv.h>
+ #include <clcmacro.h>
+
+ #ifdef cl_khr_fp16
+ #pragma OPENCL EXTENSION cl_khr_fp16 : enable
+ #endif  
+
+ double __ocml_log2_f64(double);
+ float __ocml_log2_f32(float);
+ half __ocml_log2_f16(half);
```
Comment (jchlanda) [inline libclc/amdgcn-amdhsa/libspirv/math/log2.cl:10]: Should it be hidden behind an `ifdef`?

Also: below clang format and further down commented out code.

File: libclc/amdgcn-amdhsa/libspirv/math/modf.cl [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <spirv/spirv.h>
+#include <clcmacro.h>
+
+double __ocml_modf_f64(double, double *);
+float __ocml_modf_f32(float, float *);
```
Comment (jchlanda) [inline libclc/amdgcn-amdhsa/libspirv/math/modf.cl:13]: New line, to keep it in line with other files?

File: libclc/amdgcn-amdhsa/libspirv/math/nextafter.cl [hunk 6]
```diff
@@ -0,0 +1,11 @@
+ #include <spirv/spirv.h>
+ #include <clcmacro.h>
+ 
+ double __ocml_nextafter_f64(double,double);
```
Comment (jchlanda) [inline libclc/amdgcn-amdhsa/libspirv/math/nextafter.cl:4]: clang-format here and the next line

**Comment by GYDmedwin** [comment]
Sorry，I delete my repository by mistake, and I forked again. But I don't know how to update this request now, should I close this one ,and create a new pull request?

File: libclc/amdgcn-amdhsa/libspirv/math/acos.cl [hunk 7]
```diff
@@ -0,0 +1,11 @@
+ #include <spirv/spirv.h>
```
[the remark this answers is not in the document]
Comment (GYDmedwin) [reply libclc/amdgcn-amdhsa/libspirv/math/acos.cl:1]: Sorry，I delete my repository by mistake, and I forked again. But I don't know how to update this request now, should I close this one ,and create a new pull request?

File: libclc/amdgcn-amdhsa/libspirv/math/nextafter.cl (hunk 6 above)
Comment (GYDmedwin) [reply libclc/amdgcn-amdhsa/libspirv/math/nextafter.cl:4]: Sorry，I delete my repository by mistake, and I forked again. But I don't know how to update this request now, should I close this one ,and create a new pull request?

File: libclc/amdgcn-amdhsa/libspirv/math/sincos.cl [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+ //#include <clcmacro.h>
+ 
+// double __ocml_sincos_f64(double,double);
+// float __ocml_sincos_f32(float,float);
+ 
+// #define __CLC_FUNCTION __spirv_ocl_sincos
+// #define __CLC_BUILTIN __ocml_sincos
+// #define __CLC_BUILTIN_F __CLC_XCONCAT(__CLC_BUILTIN, _f32)
+// #define __CLC_BUILTIN_D __CLC_XCONCAT(__CLC_BUILTIN, _f64)
+// #include <math/binary_builtin.inc>
```
[the remark this answers is not in the document]
Comment (GYDmedwin) [reply libclc/amdgcn-amdhsa/libspirv/math/sincos.cl:11]: Sorry，I delete my repository by mistake, and I forked again. But I don't know how to update this request now, should I close this one ,and create a new pull request?

**Pull request closed without merging**